src/control/lib/hardware/topology.go

```diff
@@ -182,6 +244,62 @@ func (t *Topology) Merge(newTopo *Topology) error {
 			}
 		}
 	}
+	return t.mergeVirtualDevices(newTopo)
+}
+
+func (t *Topology) mergeVirtualDevices(newTopo *Topology) error {
+	curDevices := t.AllDevices()
+
+	for _, newVirt := range newTopo.VirtualDevices {
+		if curDev, found := curDevices[newVirt.Name]; found {
+			curVirt, ok := curDev.(*VirtualDevice)
+			if !ok {
+				return errors.Errorf("virtual device %q has same name as a hardware device", newVirt.Name)
+			}
+
+			if newVirt.Type != DeviceTypeUnknown {
+				curVirt.Type = newVirt.Type
+			}
+
+			if err := mergeBackingDev(curDevices, curVirt, newVirt); err != nil {
```

Comment:
Same comment as for the device type.

Reply:
We return the error, so I don't think it's necessary to log it here.